core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/version/VersionCreateRequest.java

```diff
@@ -236,15 +231,43 @@ private String buildVersionUrl(RepositoryContext context, TerminologyResource re
 	}
 
 	private Map<ResourceURI, TerminologyResource> fetchResources(ServiceProvider context) {
-		return ResourceRequests.prepareSearch()
+		final Map<ResourceURI, TerminologyResource> resourcesToVersion = new HashMap<>();
+		
+		ResourceRequests.prepareSearch()
 			.one()
 			.filterById(resource.getResourceId())
 			.buildAsync()
 			.execute(context)
 			.stream()
 			.filter(TerminologyResource.class::isInstance)
 			.map(TerminologyResource.class::cast)
-			.collect(Collectors.toMap(Resource::getResourceURI, r -> r));
+			.forEach(resource -> {
+				resourcesToVersion.put(resource.getResourceURI(), resource);
+			});
+		
+		// if the resource to version is a collection URI then version all child resources as well
+		TerminologyResource terminologyResource = resourcesToVersion.get(resource);
+		if (terminologyResource instanceof TerminologyResourceCollection resourceCollection) {
+			final var registry = context.service(TerminologyResourceCollectionToolingSupport.Registry.class);
+			final Set<String> childResourceTypes = registry.getAllByToolingId(resourceCollection.getToolingId())
+				.stream()
+				.flatMap(toolingSupport -> toolingSupport.getSupportedChildResourceTypes().stream())
+				.collect(Collectors.toSet());
+			
+			ResourceRequests.prepareSearch()
+				.filterByBundleAncestorId(resourceCollection.getId())
```

Comment:
This method still makes me nervous a bit. Can we have a filterByResourceCollection helper method that internally does the very same thing, but at least the naming is more straightforward?

Reply:
I was thinking about this as well yesterday. Adding a new method for clarity.